staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/interface.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+
+	"github.com/google/cel-go/common/types/ref"
+)
+
+type PolicyDecisionKind string
+
+const (
+	Admit PolicyDecisionKind = "Admit"
+	Deny  PolicyDecisionKind = "Deny"
+)
+
+type FailurePolicy string
+
+const (
+	Fail   FailurePolicy = "Fail"
+	Ignore FailurePolicy = "Ignore"
+)
+
+type PolicyDecision struct {
+	PolicyDecisionKind `json:"kind"`
+	Message            any `json:"message"`
+}
+
+type EvaluatorFunc func(a admission.Attributes, params *unstructured.Unstructured) []PolicyDecision
+
+// Object used by admission controller to convert runtime values into CEL-typed
+// values.
+type CELObjectConverter interface {
+	// Given a GVR, look up the openapi or JSONSchemaProps, structural schema, etc.
+	// and compile it into something that can be used to turn objects into CEL
+	// values
+	DeclForResource(gvr schema.GroupVersionResource) any
+
+	// Object may be a typed native object or an unstructured object
+	ValueForObject(value any, decls any) *ref.Val
+}
+
+// A stub interface for internal policy definition type
+type fantasyInternalPolicyDefinition interface {
+	runtime.Object
+
+	// returns namespace/name
+	GetNamespace() string
+	GetName() string
+
+	// says whether this policy definition matches the provided admission resource request
+	Matches(a admission.Attributes) bool
+
+	Compile(
+		// Definition is provided with a converter which may be used by the
+		// return evaluator function to convert objects into CEL-typed objects
+		objectConverter CELObjectConverter,
+	) (EvaluatorFunc, error)
+
+	GetParamSource() schema.GroupVersionKind
+
+	GetFailurePolicy() FailurePolicy
+}
+
+type fantasyInternalPolicyBinding interface {
```

Comment:
A comment explaining why this is a fantasy would be helpful :)

Reply:
More of a holdover from prototyping days. It is an interface we expect the real internal types to conform to once they have gone thru the (possibly long) API Review. (https://github.com/kubernetes/kubernetes/pull/112883)

Using an interface this way allows us to test the component and keep building out the feature without being blocked by the real types. I've added a comment expounding upon this more, and removed the `fantasyInternal` prefix from the interfaces.